game-core/src/test/resources/Total_World_War_Dec1941.xml

```diff
@@ -7749,22 +7809,32 @@
       <option name="consumesUnits" value="1:Material"/>
       <option name="requiresUnits" value="germanCombatEngineer"/>
     </attachment>
-    <attachment name="unitAttachment" attachTo="germanInfantry" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">
+    <attachment foreach="Infantry:Players:Barracks" name="unitAttachment" attachTo="$Infantry$" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">
       <option name="movement" value="1"/>
       <option name="attack" value="2"/>
       <option name="defense" value="3"/>
       <option name="transportCost" value="2"/>
-      <option name="isInfantry" value="true"/>
+      <option name="isLandTransportable" value="true"/>
+      <option name="canBeGivenByTerritoryTo" value="$Players$"/>
+      <option name="requiresUnits" value="$Barracks$"/>
+      <option name="canInvadeOnlyFrom" value="none"/>
+    </attachment>
+    <!-- <attachment name="unitAttachment" attachTo="germanInfantry" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">
+      <option name="movement" value="1"/>
+      <option name="attack" value="2"/>
+      <option name="defense" value="3"/>
+      <option name="transportCost" value="2"/>
+      <option name="isLandTransportable" value="true"/>
       <option name="canBeGivenByTerritoryTo" value="Germany"/>
       <option name="requiresUnits" value="germanBarracks"/>
       <option name="canInvadeOnlyFrom" value="none"/>
-    </attachment>
+    </attachment> -->
     <attachment name="unitAttachment" attachTo="germanAntiTankGun" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">
       <option name="attack" value="1"/>
       <option name="defense" value="3"/>
       <option name="isAAforCombatOnly" value="true"/>
       <option name="typeAA" value="AntiTankGun"/>
-      <option name="targetsAA" value="russianMech.Infantry:russianTank:britishMech.Infantry:britishTank:americanMech.Infantry:americanTank:chineseMech.Infantry:chineseTank:spanishTank:swedishTank:turkishTank:brazilianTank"/>
+      <option name="targetsAA" value="AntiTankTargets"/>
```

Comment:
Most other templating systems have some sort of notation for variables, looking at the usage, the `AntiTankTargets` value looks identical to the literal value `AnitTankGun`.

Perhaps we should do something to make variables stand out. I'm also wondering about name collisions or similar names. What do you think @ron-murhammer ?

Thinking of other templating systems, curly braces are pretty common, we could use those, or perhaps square brackets.

Reply:
@DanVanAtta @RoiEXLab I'm fine adding an identifier symbol for variables. Open to suggestions. Using $xxx$ to identify the elements in a variable when using foreach already.

Reply:
What about using double `$` (`$$var$$`) when outside of a loop?

Reply:
@RoiEXLab @DanVanAtta So I don't really like doubling the dollar sign. Brackets/braces tend to make me think of arrays. How about using $ for variable symbol and @ for variables being looped over so examples: 
```
$variable$
@foreach_element@
```

Reply:
@ron-murhammer I'm fine with your proposal